### PR TITLE
Adding manifest files for SSMS 21.0 Preview 1.0

### PR DESCRIPTION
**Pull request by Matteo-T**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)? **Yes**
- [x] Is there a linked Issue? **No**

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

We released SSMS 21.0 Preview 1.0 today, so these are the winget manifest files.

Couple of things (vs older SSMS manifests):
- They are based on the template used by Visual Studio
- There's only one language, since the Visual Studio Installer takes care of all the languages directly
- Links, descriptions, etc... partially carried over from the old SSMS manifest files
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/190855)

### CODE REVIEW

**Review by kisantia — APPROVED** [review]

File: manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+PackageIdentifier: Microsoft.SQLServerManagementStudio.21.Preview
```
Comment (stephengillie) [inline manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.installer.yaml:2]: Please update file name and path to match this change.

File: manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.locale.1.6.0.schema.json
+PackageIdentifier: Microsoft.SQLServerManagementStudio.21.Preview
```
Comment (stephengillie) [inline manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.locale.en-US.yaml:2]: Please update file name and path to match this change.

File: manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.6.0.schema.json
```
Comment (stephengillie) [inline manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.yaml:1]: Please update file name and path to match this change.

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @Matteo-T,

Automatic Validation ended with:
> Manifest: The manifest file path must match PackageIdentifier and PackageVersion properties from manifest. actual 
'manifests\m\Microsoft\SQLServerManagementStudio\Preview\21.0', expected 
'manifests\m\Microsoft\SQLServerManagementStudio\21\Preview\21.0'

Edit: It appears that the path omits the `21`.

File: manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.yaml (hunk 3 above)
Comment (Matteo-T) [reply manifests/m/Microsoft/SQLServerManagementStudio/Preview/21.0/Microsoft.SQLServerManagementStudio.21.Preview.yaml]: I removed the (redundant and problematic) "21" from a few places... let's see how it foes this time around.

**Comment by stephengillie** [comment]
Hi @,

Do you approve of these changes?

(Automated response - build 897.)

**Review by stephengillie — APPROVED** [review]

**Pull request merged**